---
title: "Final Project"
author: "Tyler Rodrigues"
date: "2024-05-06"
output:
  html_document:
    toc: true
    toc_float:
      toc_collapsed: true
    toc_depth: 6
    theme: simplex
    code_folding: hide
bibliography: references.bib
---
\

<!-- library(usethis) -->
<!-- use_git_config(user.name = "tgrodrigues13", user.email = "tgrodrigues@umass.edu") -->

# **Bio 476 - Final Research Project**
<p style="font-size:20pt; color:black">Lower Teakettle --- Chloroflexota</p>

<p style="font-size:16pt; color:black">Tyler Rodrigues and Adam Maghraoui</p>

\

```{r, eval=F, echo=F}
#'[Loading in Libraries]

if (!require("BiocManager", quietly=TRUE))
    install.packages("BiocManager")

BiocManager::install("treeio")
BiocManager::install("ggtreeExtra")
```
```{r, echo=F}
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(knitr))
suppressPackageStartupMessages(library(ggtree))
suppressPackageStartupMessages(library(TDbook))
suppressPackageStartupMessages(library(ggimage))
suppressPackageStartupMessages(library(rphylopic))
suppressPackageStartupMessages(library(treeio))
suppressPackageStartupMessages(library(tidytree))
suppressPackageStartupMessages(library(ape))
suppressPackageStartupMessages(library(TreeTools))
suppressPackageStartupMessages(library(phytools))
suppressPackageStartupMessages(library(ggnewscale))
suppressPackageStartupMessages(library(ggtreeExtra))
suppressPackageStartupMessages(library(ggstar))
suppressPackageStartupMessages(library(plotly))
suppressPackageStartupMessages(library(zoo))
```

# Abstract
<!-- Short like 5 sentences -->
Do you ever wonder what scientists are actually doing to actively combat climate change? In this project there will be only a glimpse into some of the data that is collected on daily basis from both terrestrial and aquatic sites across the United States. The National Science Foundation's National Ecological Observatory Network (NEON) has been collecting daily data since 2019 on every organism possible. Metagenomic data and chemical properties data are collected and put into an open access database that anyone can use and learn from. In this project we aim to look at this data and practice our data wrangling, analysis, and presentation skills. Taking a deeper look at the phylogenies of the Lower Teakettle Site in California or the Chloroflexota phylum, as well as graphs showing things such as 16s rRNA counts or bin completion/contamination will allow us, and other scientists, to learn from this data and act accordingly by making plans for future change. 

\

# Motivating Reasons
<!-- Probably 1 paragraph -->
This NEON project is incredibly important as it is serving as an open source for all kinds of data that scientists need in order to track how the environment and climate is changing. The ideas behind NEON began to come into place right around 2000 as scientists began to gather and collectively decided that an ecological observatory network across the continental United States, Hawaii, and Puerto Rico was a necessity if they were to try and battle ongoing climate change. The official NEON plan was not finalized until 2006, and funding was not available until 2011. Construction and setup began soon after and operations launched in the summer of 2019 and has running and collecting fantastic data ever since. Motivation from many scientists and researchers involved keep this study alive and it is their dedication that may eventually allow us to fix some of the bigger issues with climate change. For personal motivation we are biology majors who care deeply about the world as well and want to make a difference. Everybody, from the equipment mechanics to the guy behind the screen playing with the data (us), play a humongous part in all the research that gets done. Hopefully one day in our lifetime we can take part in making a change and finally making a dent in this constant problem of climate change. 

\

# Introduction
<!-- Probably 2 paragraphs -->
The National Science Foundation's National Ecological Observatory Network (NEON) is a country wide program operated by Battelle and is set up to collect long term ecological data. This is an open access data base with the goal of better understanding how nature and the ecosystems are constantly changing. In our specific project we aim to focus in on one of these NEON sites as well as a specific phylum. We will take a deep dive into phylogenies, metagenome-assembled (MAG) data, metagenomic data and chemistry data to gain a deeper understanding of these species and their ecosystems. We specifically will look in on the Lower Teakettle region located at Sierra National Forest in California, USA as well as the Chloroflexota phylum. 

\

![Lower Teakettle, Sierra Nevada Forest, California, USA](images/Lab 10 - NEON Project Intro Picture.jpg){width=300px, height=400px}

\

The Lower Teakettle NEON site is an evergreen forest biome that spans for 51.4 km^2 throughout the Sierra National Forest in California. With hot and dry summers and mild, wet winters, and a mean annual temperature of 8Â°C, this site is home to thousands of varying organisms and vegetation. Dominated by red and white firs, pines, and plenty of shrubbery, organisms can thrive. With varying elevation from 1900-2807 meters and a mixed terrain of mountains, meadows, and high granite peaks, this makes the TEAK site the perfect area for a long-term ecological study. This site makes up 1 of the 4 located in what is known as the Pacific Southwest Domain (sector D17). Aquatic site Teakettle 2 Creek is colocated with this terrestrial site, and together these sites can produce data on any species or ecosystem present in the area. 

[@noauthor_nsf_nodate]

Chloroflexota is a phylum created first in 2001 in Volume 1 of Bergey's Manual of Systematic Bacteriology, and it contains bacteria that are monoderms (having one cell membrane with no outer membrane). There are a vast group of bacteria in this phylum from aerobic thermophiles to anoxygenic phototrophs. ORganisms now found within the phylum Chloroflexota were at one point grouped together with some of its closest relatives from 3 different generas titled 'Chloroflexus', 'Herpetosiphon', and 'Thermomicrobium' into a grouped genera titled 'Chloroflexi' (a.k.a. Chloroflexota). This grouping was made in 1987 by Carl Woese, where he studied the 16s rRNA counts to distinguish between organisms. Ever since 2001 however, this groupings of genera have been separated and the phylum Chloroflexota is now divided into several classes. Recent studies published in a Dutch microbiology journal show that scientists want to further split the Chloroflexota phylum as many of the classes that make up the phylum have weak evidence of grouping through a phylogeny. The two classes of 'Chloroflexi' and 'Thermomicrobia'are found to share similarities in the some 50s ribosomal proteins, leading scientists to believe that these two classes only make up the phylum Chloroflexota. 

[@noauthor_chloroflexota_2024]
\
[@gupta_phylogenetic_2013]
\
[@wiegand_taxonomic_2023]

\

# Methods
```{r, message=F, warning=F}
#'[MAG Data]

NEON_MAGs <- read_csv("data/GOLD_Study_ID_Gs0161344_NEON_2024_4_21.csv") %>% 
#remove columns that are not needed for data analysis
  select(-c(`GOLD Study ID`, `Bin Methods`, `Created By`, `Date Added`, `Bin Lineage`)) %>% 
#create a new column with the Assembly Type
  mutate("Assembly Type" = case_when(`Genome Name` == "NEON combined assembly" ~ `Genome Name`, TRUE ~ "Individual")) %>% 
  mutate_at("Assembly Type", str_replace, "NEON combined assembly", "Combined") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "d__", "") %>%  
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "p__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "c__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "o__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "f__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "g__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "s__", "") %>%
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), ";", remove=FALSE) %>%
  mutate_at("Domain", na_if,"") %>% 
  mutate_at("Phylum", na_if,"") %>% 
  mutate_at("Class", na_if,"") %>% 
  mutate_at("Order", na_if,"") %>% 
  mutate_at("Family", na_if,"") %>% 
  mutate_at("Genus", na_if,"") %>% 
  mutate_at("Species", na_if,"") %>% 
#Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
#Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
#Get rid of the the common string "S-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
#separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
#separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-")
```
```{r, message=F, warning=F}
#'[Metagenome Data]

NEON_metagenomes <- read_tsv("data/exported_img_data_Gs0161344_NEON.tsv") %>% 
  select(-c(`Domain`, `Sequencing Status`, `Sequencing Center`)) %>% 
  rename(`Genome Name` = `Genome Name / Sample Name`) %>% 
  filter(str_detect(`Genome Name`, 're-annotation', negate = T)) %>% 
  filter(str_detect(`Genome Name`, 'WREF plot', negate = T))

#'[Reformatting Genome Name]

NEON_metagenomes <- NEON_metagenomes %>% 
#Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
#Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
#Get rid of the the common string "-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
#separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
#separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-")
```
```{r, message=F}
#'[Chemistry Data]

NEON_chemistry <- read_tsv("data/neon_plot_soilChem1_metadata.tsv") %>% 
#remove -COMP from genomicsSampleID
  mutate_at("genomicsSampleID", str_replace, "-COMP", "")
```
```{r}
#'[Joining MAG, Metagenome, and Chemistry into a Single Dataframe]

NEON_combo <- NEON_MAGs %>% 
  left_join(NEON_metagenomes, by = "Sample Name") %>% 
  left_join(NEON_chemistry, by = c("Sample Name" = "genomicsSampleID")) %>% 
  rename("label" = "Bin ID")
```
```{r, echo=F}
#'[Filter Combo Data to Taxonomic Group or Site]

NEON_combo_Chloroflexota <- NEON_combo %>% 
  filter(Phylum == "Chloroflexota")

NEON_combo_TEAK <- NEON_combo %>% 
  filter(`Site ID.x` == "TEAK") %>% 
  filter(Domain == "Bacteria")
```
```{r}
#'[Creating Individaual and Co Data Objects]

NEON_MAGs_Ind <- NEON_MAGs %>%
  filter(Domain == "Bacteria") %>%
  filter(`Assembly Type` == "Individual")

# NEON_MAGs_Co <- NEON_MAGs %>% filter(Domain == "Bacteria") %>% filter(`Assembly Type` == "Combined")
```
```{r, warning=F, message=F}
#'[Loading in Sankey NEON Data]

NEON_MAGs_Sankey <- read_csv("data/GOLD_Study_ID_Gs0161344_NEON_edArchaea.csv") %>%
#remove columns that are not needed for data analysis
  select(-c(`GOLD Study ID`, `Bin Methods`, `Created By`, `Date Added`)) %>%
#create a new column with the Assembly Type
  mutate("Assembly Type" = case_when(`Genome Name` == "NEON combined assembly" ~ `Genome Name`, TRUE ~ "Individual")) %>%
  mutate_at("Assembly Type", str_replace, "NEON combined assembly", "Combined") %>%
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), "; ", remove = FALSE) %>%
#Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>%
#Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>%
#Get rid of the the common string "S-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
#separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>%
#separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-")
```
\

Above, hidden, is where MAG data, metagenome data, and chemistry data were all read in and formatted into a proper working manner. Data for phylogeny trees and Sankey plots is also read in and transformed into the desired form. 

The NEON experiment is a huge long term experiment, and this means that they have set established methods for each region and for each test that is needed to be ran. Across all sites, NEON runs airborne remote sensing every 2-5 years, automated instrument testing that collects data year round, and observational sampling conducted by field scientists during growing seasons. Airborne remote sensing allows the scientists to get a view of the site from the air to provide updates on canopy cover and big changes to the tree line. Airplanes also use lidar to gain a 3D view of the landscape as well as a spectrometer to look at land cover types and vegetation content. Some automated instruments found on site include a meterorological tower to collect the local weather and climate data, a soil sensor array to measure chemical properties of soil, and phenocams are used to caputre changes in foliage and canopy from both above and below the tree line. Observational sampling is the most hands-on as it requires scientists to be physically present at the site where they collect samples of anything they want to further study back in the lab. Instrument maintenence is usually also done during this time if need be. 

[@hofmeyr_terabase-scale_nodate] 
\
[@clum_doe_2021]
\
[@noauthor_jgi_nodate]
\
[@noauthor_doe_nodate]

\

# Results

### Site Graphs - Lower Teakettle, Sierra National Forest, California, USA

<!-- Circle Phylogeny for TEAK With Colored Phylum Labels -->

```{r, echo=F}
#'[Reading in Tree Files from GTDB]

#tree_arc <- read.tree("data/gtdbtk.ar53.decorated.tree")
tree_bac <- read.tree("data/gtdbtk.bac120.decorated.tree")
```
```{r, results=F, echo=F}
#'[Getting Your Subtree]

#Make a vector with the internal node labels
node_vector_bac = c(tree_bac$tip.label,tree_bac$node.label)

#Search for your Phylum or Class to get the node
grep("Chloroflexota", node_vector_bac, value = TRUE)
```
```{r, results=F, echo=F}
match(grep("Chloroflexota", node_vector_bac, value = TRUE), node_vector_bac)
```
```{r, echo=F}
#'[Need to Preorder Tree Before Extracting]

tree_bac_preorder <- Preorder(tree_bac)
tree_Chloroflexota <- Subtree(tree_bac_preorder, 1712)
```
```{r, fig.width = 10, fig.height = 8, fig.align = 'center'}
#'[Graph Chunk]

TEAK_MAGs_label <- NEON_combo_TEAK$label

tree_bac_TEAK_MAGs <-drop.tip(tree_bac,tree_bac$tip.label[-match(TEAK_MAGs_label, tree_bac$tip.label)])

ggtree(tree_bac_TEAK_MAGs, layout="circular") %<+%
  NEON_combo +
  geom_point(mapping=aes(fill=Phylum), size = 2.5, shape = 23) +
  ggtitle("Different Phylums Located at Lower Teakettle") +
  theme(plot.title = element_text(hjust = 0.5),(size = 30)) #face = "bold"
```
This phylogeny circle shows the different species located at the Lower Teakettle Site. It is always important for scientists to know what organisms we are working with, and there is no better way than a phylogeny to show that. The phylogeny is color coded by phylum. Note that a solution to the grey 'NA' nodes was never found. 

<!-- MAGs at Our Site Colored by Phylum -->

```{r, results = "hold", fig.width = 8, fig.height = 4}
#'[Table and Graph Chunk]

NEON_MAGs_Site <- NEON_MAGs %>% 
  filter(Domain == "Bacteria") %>% 
  filter(`Site ID` == "TEAK") %>% 
  select(c(`Site ID`, `Domain`, `Phylum`, `Class`, `Order`, `Family`, `Genus`))

kable(NEON_MAGs_Site)

NEON_MAGs_Site %>% 
  ggplot(aes(x = fct_rev(fct_infreq(`Site ID`)), fill = Phylum)) +
  geom_bar(position = position_dodge2(width = 0.8, preserve = "single")) +
  labs(title = "MAG Counts at Lower Teakettle", x = "Site ID", y = "MAG Count")
```

<!-- Scatter for TEAK showing 16s RNA Count -->

```{r, fig.width = 10, fig.height = 6, fig.align = 'center'}
#'[Graphing Chunk]

NEON_combo_TEAK %>% 
  ggplot(aes(x=`Total Number of Bases`, y = `Bin Completeness`, color = `16s rRNA`)) +
  geom_point() +
  theme(axis.text.x = element_text(color = "black", angle = 40, hjust = 1, vjust = 1, size = 10)) +
  labs(title = "Total Number of Bases vs Bin Completeness at Lower Teakettle Site", x = "Total Number of Bases", y = "Bin Completeness (%)", color = "Number of 16s rRNA")
```

This scatter plot shows data from the Lower Teakettle site and compares total number of bases to bin completion. The number of 16s RNA per organism is also highlighted.  

<!-- Novel Bacteria for TEAK by Phylum -->

```{r, fig.width = 10, fig.height = 6}
#'[Graphing Chunk]

NEON_MAGs_Ind %>%
  filter(`Site ID` == "TEAK") %>% 
  group_by(Phylum) %>%
  summarize(novel_bacteria_count = sum(is.na(Genus))) %>%
  ggplot(aes(x = Phylum, y = novel_bacteria_count)) +
  geom_col() +
  labs(title = "Count of Novel Bacteria per Site", x = "Site ID", y = "Count of Novel Bacteria") +
  theme(axis.text.x = element_text(angle = 30, hjust = 1, size=12))
```

This graph shows the novel bacteria discovered in every individual site in the NEON database. 

<!-- High vs Low Quality Bins for TEAK -->

```{r, fig.width = 10, fig.height = 6}
#'[Graphing Chunk]

NEON_MAGs_Ind %>%
  filter(is.na(Genus)) %>%
  filter(Site == "Lower Teakettle, Sierra National Forest, California, USA") %>%
  count(`Bin Quality`) %>%
  mutate(Bin_Quality = ifelse(`Bin Quality` %in% c("MQ", "HQ"), `Bin Quality`, "NA")) %>%
  ggplot(aes(x = Bin_Quality, y = n)) +
  geom_col() +
  labs(title = "Count of Bin Quality for Unidentified Bacteria", x = "Bin Quality", y = "Count")
```
This graph shows high quality versus medium quality for our site. 

\

### Phylum Graphs - Chloroflexota

<!-- Sankey Plot of Individual Assembly at TEAK -->

```{r, warning=F}
#'[Sorting for Phylum - Individual Assembly]

NEON_MAGs_Ind_Sankey <- NEON_MAGs_Sankey %>%
  filter(`Assembly Type` == "Individual")

NEON_MAGs_Ind_Chloro_Sankey <- NEON_MAGs_Ind_Sankey %>%
  filter(Phylum == "Chloroflexota")

#Select the GTDB Taxonomic lineage and separate into taxonomic levels
sankey_data <- NEON_MAGs_Ind_Chloro_Sankey %>%
  select(`GTDB-Tk Taxonomy Lineage`) %>%
#NAs are likely Archaea
  replace_na(list(`GTDB-Tk Taxonomy Lineage` = 'Archaea')) %>%
#Pavian format requires p__ etc
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), "; ")

#Fill in the NAs with the taxonomic the higher taxonomic level to the left
sankey_data[] <- t(apply(sankey_data, 1, zoo::na.locf))

#Put the data into a format that can be read by the Sankey App
sankey_data <- sankey_data %>%
  unite(col = "classification", c(Domain, Phylum, Class, Order, Family, Genus, Species), sep='; ') %>%
  mutate_at("classification", str_replace, "Archaea", "d__Archaea") %>%
  mutate_at("classification", str_replace, "Bacteria", "d__Bacteria") %>%  
  mutate_at("classification", str_replace, "; ", "|p__") %>%
  mutate_at("classification", str_replace, "; ", "|c__") %>%
  mutate_at("classification", str_replace, "; ", "|o__") %>%
  mutate_at("classification", str_replace, "; ", "|f__") %>%
  mutate_at("classification", str_replace, "; ", "|g__") %>%
  mutate_at("classification", str_replace, "; ", "|s__")  

#Create format for Pavian with counts for each taxonomic level
sankey_data_s <- sankey_data
sankey_data_g <- sankey_data
sankey_data_f <- sankey_data
sankey_data_o <- sankey_data
sankey_data_c <- sankey_data
sankey_data_p <- sankey_data
sankey_data_d <- sankey_data

sankey_data_g$classification <- sub("\\|s__.*", "", sankey_data_g$classification)  
sankey_data_f$classification <- sub("\\|g__.*", "", sankey_data_f$classification)  
sankey_data_o$classification <- sub("\\|f__.*", "", sankey_data_o$classification)  
sankey_data_c$classification <- sub("\\|o__.*", "", sankey_data_c$classification)  
sankey_data_p$classification <- sub("\\|c__.*", "", sankey_data_p$classification)  
sankey_data_d$classification <- sub("\\|p__.*", "", sankey_data_d$classification)  

sankey_data_allTaxa <- bind_rows(sankey_data_s, sankey_data_g, sankey_data_f, sankey_data_o, sankey_data_c, sankey_data_p, sankey_data_d) %>%
  mutate(classification = as.factor(classification)) %>%
  count(classification) %>%
#rename for Pavian format
  rename(`#SampleID` = `classification`) %>%
  rename(`Metaphlan2_Analysis` = `n`)

#Write file to input to Pavian Sankey
write_tsv(sankey_data_allTaxa, "Ind_Chloro.txt")
```
```{r}
knitr::include_url("sankey-Ind_Chloro.txt.html")
```
This Sankey plot shows the taxononomy lineage of our phylum, Chloroflexota. This Sankey plot is using individual assemblies only. The provided code filters and prepares taxonomic lineage data from NEON MAGs, formats it for a Sankey diagram, and saves it in a format suitable for the Pavian tool, which is used to visualize metagenomic classifications. 

<!-- Sankey Plot of Combined Assembly at TEAK -->

```{r, warning=F}
#'[Sorting for Phylum - Combined Assembly]

NEON_MAGs_Co_Sankey <- NEON_MAGs_Sankey %>%
  filter(`Assembly Type` == "Combined")

NEON_MAGs_Co_Chloro_Sankey <- NEON_MAGs_Co_Sankey %>%
  filter(Phylum == "Chloroflexota")

#Select the GTDB Taxonomic lineage and separate into taxonomic levels
sankey_data <- NEON_MAGs_Co_Chloro_Sankey %>%
  select(`GTDB-Tk Taxonomy Lineage`) %>%
#NAs are likely Archaea
  replace_na(list(`GTDB-Tk Taxonomy Lineage` = 'Archaea')) %>%
#Pavian format requires p__ etc
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), "; ")

#Fill in the NAs with the taxonomic the higher taxonomic level to the left
sankey_data[] <- t(apply(sankey_data, 1, zoo::na.locf))

#Put the data into a format that can be read by the Sankey App
sankey_data <- sankey_data %>%
  unite(col = "classification", c(Domain, Phylum, Class, Order, Family, Genus, Species), sep='; ') %>%
  mutate_at("classification", str_replace, "Archaea", "d__Archaea") %>%
  mutate_at("classification", str_replace, "Bacteria", "d__Bacteria") %>%  
  mutate_at("classification", str_replace, "; ", "|p__") %>%
  mutate_at("classification", str_replace, "; ", "|c__") %>%
  mutate_at("classification", str_replace, "; ", "|o__") %>%
  mutate_at("classification", str_replace, "; ", "|f__") %>%
  mutate_at("classification", str_replace, "; ", "|g__") %>%
  mutate_at("classification", str_replace, "; ", "|s__")  

#Create format for Pavian with counts for each taxonomic level
sankey_data_s <- sankey_data
sankey_data_g <- sankey_data
sankey_data_f <- sankey_data
sankey_data_o <- sankey_data
sankey_data_c <- sankey_data
sankey_data_p <- sankey_data
sankey_data_d <- sankey_data

sankey_data_g$classification <- sub("\\|s__.*", "", sankey_data_g$classification)  
sankey_data_f$classification <- sub("\\|g__.*", "", sankey_data_f$classification)  
sankey_data_o$classification <- sub("\\|f__.*", "", sankey_data_o$classification)  
sankey_data_c$classification <- sub("\\|o__.*", "", sankey_data_c$classification)  
sankey_data_p$classification <- sub("\\|c__.*", "", sankey_data_p$classification)  
sankey_data_d$classification <- sub("\\|p__.*", "", sankey_data_d$classification)  

sankey_data_allTaxa <- bind_rows(sankey_data_s, sankey_data_g, sankey_data_f, sankey_data_o, sankey_data_c, sankey_data_p, sankey_data_d) %>%
  mutate(classification = as.factor(classification)) %>%
  count(classification) %>%
#rename for Pavian format
  rename(`#SampleID` = `classification`) %>%
  rename(`Metaphlan2_Analysis` = `n`)

#Write file to input to Pavian Sankey
write_tsv(sankey_data_allTaxa, "Co_Chloro_pavian.txt")
```
```{r}
knitr::include_url("sankey-Co_Chloro_pavian.txt.html")
```
This Sankey plot visualizes the taxonomic lineage of the phylum Chloroflexota, based on combined assembly data only. 

<!-- Bar Plot of Phylum Across All Sites -->

```{r, results = "hold", fig.width = 10, fig.height = 4}
NEON_MAGs_NoNA <- NEON_MAGs %>% 
  filter(`Site ID` != "NA")

kable(NEON_MAGs_NoNA %>% 
  select(c(`Site ID`, `Domain`, `Phylum`, `Class`, `Order`, `Family`, `Genus`)) %>% 
  filter(`Phylum` == "Chloroflexota"))

NEON_MAGs_NoNA %>% 
  ggplot(aes(x = (Phylum = "Chloroflexota"), fill = `Site ID`)) +
  geom_bar(position = position_dodge2(width = 0.8, preserve = "single")) +
  labs(title = "MAG Counts of Chloroflexota by Site", x = "Phylum", y = "MAG Count")
```

<!-- Triple Graph of Phylum Showing Bin Completeness/Contamination -->

```{r}
#For unknown reasons the following does not like blank spaces in the names
NEON_combo_noblank <- NEON_combo %>% 
  rename("AssemblyType" = "Assembly Type") %>% 
  rename("BinCompleteness" = "Bin Completeness") %>% 
  rename("BinContamination" = "Bin Contamination") %>% 
  rename("TotalNumberofBases" = "Total Number of Bases") %>% 
  rename("EcosystemSubtype" = "Ecosystem Subtype") 

ggtree(tree_Chloroflexota)  %<+%
  NEON_combo + 
  geom_tippoint(aes(colour=`Site ID.x`)) + 
#For unknown reasons the following does not like blank spaces in the names
  geom_facet(panel = "Bin Completeness", data = NEON_combo_noblank, geom = geom_point, mapping=aes(x = BinCompleteness)) +
  geom_facet(panel = "Bin Contamination", data = NEON_combo_noblank, geom = geom_col, aes(x = BinContamination), orientation = 'y', width = .6) +
  theme_tree2(legend.position=c(.1, .7)) +
  ggtitle("Tri-Graph of Chloroflexota Phylogeny and Bin Completion/Contamination") +
  theme(plot.title = element_text(hjust = 0.5),(size = 30)) + #face = "bold"
  theme(legend.position.inside = c(0.05, 0.4), legend.background = element_rect(fill = "white", colour = NA)) # c(0,0) bottom left, c(1,1) top-right
```

These three graphs together show the phylogeny of Chloroflexota along with both individual and combined assembly data for Bin Completeness/Contamination. The phylogenetic tree is also highlighted with Site ID. 

<!-- Soil Temperature of Chloroflexota Across Sites -->
```{r, warning=F}
NEON_combo_Chloroflexota %>% 
  ggplot(aes(x=`Site ID.x`, y = soilTemp)) +
  geom_boxplot() +
  labs(title = "Soil Temperature by Site for Chloroflexota", x = "Site", y = "Soil Temperature (Â°C)")
```

This graph displays the distribution of soil temperatures for each sample at various sites, specifically focusing on the phylum Chloroflexota. By utilizing a boxplot, it provides a clear visualization of the temperature range, median, and variability at each site. Understanding soil temperature variations is crucial as it directly affects the growth and metabolic activities of Chloroflexota. Different temperature conditions can influence the abundance, diversity, and function of this phylum in soil environments. 

<!-- Soil pH vs Vegetation Type -->

```{r, warning=F}
NEON_combo_Chloroflexota %>% 
  ggplot(aes(x=`nlcdClass`, y = soilInCaClpH, color = Class)) +
  geom_point() +
  theme(axis.text.x = element_text(angle=45, vjust=1, hjust=1, size=9)) +
  labs(title = "Soil pH by Vegetation Type for Chloroflexota", x = "Vegetation Type", y = "pH Level")
```

This graph illustrates the relationship between soil pH in CaCl (soilInCaClpH) and land cover classification (nlcdClass) for samples in the phylum Chloroflexota. Each point represents a sample, with different colors indicating the Class order within Chloroflexota.

\

# Discussion

These graphs shown above provide a comprehensive view of the phylogeny of Chloroflexota as well as the phylogeny of the Lower Teakettle Site. Included as well are graphs showing the quality of assembly data, including bin completeness/contamination, and total number of bases. The phylogeny circle graph offers a detailed visualization of the different species the Lower Teakettle Site, color-coded by phylum. This is essential for understanding the diversity and evolutionary relationships of these organisms. The scatter plot, which compares the total number of bases to bin completion and highlights the number of 16s RNA per organism, is particularly useful. 16s RNA is an important metric as it is what allows scientists to differentiate between organisms at the genus level. Across all the major phyla of bacteria, 16s RNA has been helping scientists become more precise when developing phylogenies and studying relationships between these organisms. This kind of detailed genetic information is indispensable for developing a nuanced understanding of microbial communities. Bin completion/contamination data is important in order to see which bins may have too much contamination to use in studies, as well as keeping track of which bins are still in need of completion. As this project has been happening for a while, and will continue on for a while, it is incredibly important to keep the data and yourself as the scientist completely organized. This ensures that a quality study is done and all points of interest are covered by the study.

Furthermore, the distinction between high-quality and medium-quality data is highlighted in the graph comparing these standards for our site. High-quality data ensures more reliable and comprehensive insights, leading to robust conclusions. Conversely, medium-quality data, while still useful, might lead to less definitive insights due to its limitations. Recognizing these differences is crucial for evaluating the reliability of research findings. Additionally, the two Sankey plots visualize the taxonomic lineage of Chloroflexota, formatted for use with the Pavian tool, which aids in visualizing metagenomic classifications. This helps in understanding the genetic insights from metagenomics and establishes genomic connections between function and phylogeny in uncultured organisms. Discovering novel bacteria at each site reveals the unique microbial diversity and local adaptations present in different environments. This has significant implications for biotechnology, offering potential new enzymes and biochemical pathways. It also enhances our understanding of microbial ecology and evolution, providing insights into ecosystem functions and evolutionary processes. Additionally, these discoveries enrich microbial databases, improving the accuracy of microbial identification and classification tools.

By analyzing both individual and combined assemblies, researchers can gain a comprehensive understanding of the diversity and evolutionary relationships within the Chloroflexota phylum. This combined data approach allows for a broader view of taxonomic distribution and functional potential across different environments, providing a holistic picture of the phylum's role in various ecosystems. These visualizations and data analyses are crucial for organizing ongoing and future research, ensuring the quality and thoroughness of studies focused on Chloroflexota. Ultimately, this leads to a well-rounded understanding of the phylum's ecological significance and its potential applications in biotechnology and environmental management.

\

# Conclusion

After completing this project we have began to deep dive into the world of bioinformatics and have learned a great deal about the Lower Teakettle NEON site as well as the Chloroflexota phylum. We have demostrated and discussed how the NEON project, and data wrangling projects like these, are imporant for scientists and researchers. These projects allow massive amounts of raw data to be compiled, wrangled, and then cleaned up to provide a nice output for whoever the audience may be. This project has not only taught us a lot about the uses of programming and R in biology, but also how many fields can mix and work together to achieve great things. 

\

# References